common/djangoapps/course_modes/models.py

```diff
@@ -202,6 +202,9 @@ def save(self, force_insert=False, force_update=False, using=None):
         # Ensure currency is always lowercase.
         self.clean()  # ensure object-level validation is performed before we save.
         self.currency = self.currency.lower()
+        if self.id is None:
+            # If this model has no primary key at save time, it needs to be force-inserted.
+            force_insert = True
         super(CourseMode, self).save(force_insert, force_update, using)
```

Comment:
Or this code, which is related to the above code. All these changes were necessary to get a particular course_modes test to pass.